File: leetcode/2843. Count Symmetric Integers/main.go
package main

func countSymmetricIntegers(low int, high int) int {
	var result int

	for number := low; number <= high; number++ {
		digits := getDigits(number)
		if len(digits)%2 == 1 {
			continue
		}

		if isSymmetric(digits) {
			result++
		}
	}

	return result
}

func getDigits(number int) []int {
	var digits []int
	for number > 0 {
		digits = append(digits, number%10)
		number /= 10
	}

	for i, j := 0, len(digits)-1; i < j; i, j = i+1, j-1 {
		digits[i], digits[j] = digits[j], digits[i]
	}

	return digits
}

func isSymmetric(digits []int) bool {
	n := len(digits) / 2

	var sum1, sum2 int
	for i := range n {
		sum1 += digits[i]
		sum2 += digits[i+n]
	}

	return sum1 == sum2
}

File: leetcode/2843. Count Symmetric Integers/main.rs
impl Solution {
    pub fn count_symmetric_integers(low: i32, high: i32) -> i32 {
        let mut result = 0;

        for number in low..=high {
            let digits = Self::get_digits(number);
            if digits.len() % 2 == 1 {
                continue;
            }

            if Self::is_symmetric(&digits) {
                result += 1
            }
        }

        result
    }

    fn get_digits(mut number: i32) -> Vec<i32> {
        let mut digits = Vec::new();

        while number > 0 {
            digits.push(number % 10);
            number /= 10;
        }

        digits.reverse();

        digits
    }

    fn is_symmetric(digits: &[i32]) -> bool {
        let n = digits.len() / 2;

        let (first, second) = digits.split_at(n);

        let sum1: i32 = first.iter().sum();
        let sum2: i32 = second.iter().sum();

        sum1 == sum2
    }
}
